tools/tsdb-index/main.go

```diff
@@ -76,15 +76,16 @@ func printBlockIndex(blockDir string, printChunks bool, matchers []*labels.Match
 		return
 	}
 
+	var builder labels.ScratchBuilder
 	for p.Next() {
-		lbls := labels.Labels(nil)
 		chks := []chunks.Meta(nil)
-		err := idx.Series(p.At(), &lbls, &chks)
+		err := idx.Series(p.At(), &builder, &chks)
 		if err != nil {
 			level.Error(logger).Log("msg", "error getting series", "seriesID", p.At(), "err", err)
 			continue
 		}
 
+		lbls := builder.Labels()
```

Comment:
Shouldn't be `IndexReader.Series()` responsability to return valid labels? Labels are expected to be sorted. If input labels are not sorted, then who use the builder should call sort, otherwise it's not required. I think the expectation from the caller perspective is that whenever labels are generated (`builder.Labels()`) then they must be guaranteed to be sorted.

Reply:
That would be my expectation too, and I would drop the sorting https://github.com/grafana/mimir/blob/873839199d82636f79d46328de43b786ccc72bf3/pkg/storage/tsdb/block/index.go#L599-L600

Reply:
I agree we don't need this sort, but don't agree to make that change in this PR; make it separately and leave this PR as a vendor update.